src/test/java/com/google/sps/CommentRetrievalTest.java

```diff
@@ -0,0 +1,138 @@
+  private final int EXCESS_HUNDRED = 200;
+  private final int MAX_COMMENTS_PER_TOKEN = 100;
+  private YouTubeCommentRetriever commentRetriever;
+  private final String videoUrl = "dummyVideo";
+  private final String NEXT_PAGE_TOKEN = "Some Page Token";
+  private final String FIRST_TOKEN_COMMENT = "First";
+  private final String SECOND_TOKEN_COMMENT = "Second";
+
+  // Simulate a CommentThreadListResponse with exactly as many comments as expected
+  public void setUpYouTubeMocks(int numExpectedComments) throws Exception {
+    YouTube mockedYoutube = mock(YouTube.class, RETURNS_DEEP_STUBS);
+    YouTube.CommentThreads.List mockedCommentThreadList =
+        mock(YouTube.CommentThreads.List.class, RETURNS_DEEP_STUBS);
+    boolean useNextPageToken = (numExpectedComments > MAX_COMMENTS_PER_TOKEN);
+    // This is only enough for two calls, all these tests require
+    // Return on the first call either 100 comments or the desired number.
+    // If there's no next page token, return nothing on the second call,
+    // Otherwise return the remainder of comments
+    CommentThreadListResponse firstResponse =
+        mockThreadListResponse(
+            Math.min(numExpectedComments, MAX_COMMENTS_PER_TOKEN),
+            (useNextPageToken) ? NEXT_PAGE_TOKEN : null,
+            FIRST_TOKEN_COMMENT);
+    if (useNextPageToken) {
+      CommentThreadListResponse secondResponse =
+          mockThreadListResponse(
+              Math.min(numExpectedComments - MAX_COMMENTS_PER_TOKEN, MAX_COMMENTS_PER_TOKEN),
+              NEXT_PAGE_TOKEN,
+              SECOND_TOKEN_COMMENT);
+      when(mockedCommentThreadList.execute()).thenReturn(firstResponse, secondResponse);
+    } else {
+      when(mockedCommentThreadList.execute()).thenReturn(firstResponse);
+    }
+
+    when(mockedYoutube.commentThreads().list(anyString())).thenReturn(mockedCommentThreadList);
+    commentRetriever = new YouTubeCommentRetriever(mockedYoutube);
+  }
+
+  // Creates list of comments of desired length.
+  // Page token is needed for iteration but its contents are irrelevant.
+  private CommentThreadListResponse mockThreadListResponse(
+      int desiredComments, String nextPageToken, String commentContent) {
+    CommentThread mockThread = dummyCommentThread(commentContent);
+    return new CommentThreadListResponse()
+        .setItems(Collections.nCopies(desiredComments, mockThread))
+        .setNextPageToken(nextPageToken);
+  }
+
+  private CommentThread dummyCommentThread(String comment) {
+    // This has to be done because for some reason it's impossible to directly change a comment's
+    // content
+    CommentThread mockedCommentThread = mock(CommentThread.class, RETURNS_DEEP_STUBS);
+    when(mockedCommentThread.getSnippet().getTopLevelComment().getSnippet().getTextDisplay())
+        .thenReturn(comment);
+    return mockedCommentThread;
+  }
+
+  // The simplest case: extract 100 comments from a video with more than 100 comments
```

Comment:
Remove "the simplest case", and change the name of the test from default behavior, as it is not necessarily the default behavior.
Also in the comment you mentioned extract 100 comments from more than 100 comments, but it seems like the video only has 100 comments, which is not consistent with the comment.